plugin/evm/acp176/acp176.go

```diff
@@ -0,0 +1,197 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// ACP176 implements the fee logic specified here:
+// https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/176-dynamic-evm-gas-limit-and-price-discovery-updates/README.md
+package acp176
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"sort"
+
+	"github.com/ava-labs/avalanchego/vms/components/gas"
+	"github.com/holiman/uint256"
+
+	safemath "github.com/ava-labs/avalanchego/utils/math"
+)
+
+const (
+	MinTargetPerSecond  = 1_000_000                                 // P
+	TargetConversion    = MaxTargetChangeRate * MaxTargetExcessDiff // D
+	MaxTargetExcessDiff = 1 << 15                                   // Q
+	MinBaseFee          = 1                                         // M
+
+	TimeToFillCapacity            = 10   // in seconds
+	TargetToMax                   = 2    // multiplier to convert from target per second to max per second
+	TargetToPriceUpdateConversion = 43   // 43s ~= 30s * ln(2) which makes the price double at most every ~30 seconds
+	MaxTargetChangeRate           = 1024 // Controls the rate that the target can change per block.
+
+	targetToMaxCapacity = TargetToMax * TimeToFillCapacity
+	maxTargetExcess     = 1_024_950_627 // TargetConversion * ln(MaxUint64 / MinTargetPerSecond) + 1
+)
+
+var minBaseFee = big.NewInt(MinBaseFee)
+
+// State represents the current state of the gas pricing and constraints.
+type State struct {
+	Gas          gas.State
+	TargetExcess gas.Gas
```

Comment:
`TargetExcess` is the latest value of $q$ right? I think I'm missing something...could you explain the rationale for including "excess" in the naming convention here? It initially had me confused with the excess gas used over the current target.